src/evict/evict_page.c

```diff
@@ -268,9 +268,16 @@ __evict_page_clean_update(WT_SESSION_IMPL *session, WT_REF *ref, bool closing)
 	 * Discard the page and update the reference structure; if the page has
 	 * an address, it's a disk page; if it has no address, it's a deleted
 	 * page re-instantiated (for example, by searching) and never written.
+	 *
+	 * If evicting a WT_REF_LIMBO reference, we get to here and transition
+	 * back to WT_REF_LOOKASIDE, else discarding any lookaside information.
 	 */
 	__wt_ref_out(session, ref);
-	if (ref->addr == NULL) {
+	if (!closing && ref->page_las != NULL &&
+	    ref->page_las->eviction_to_lookaside == true) {
```

Comment:
Style nit: should just be `&& ref->page_las->eviction_to_lookaside`: avoid equality comparison with `true` or `false`.